oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/web/authentication/DefaultAuthenticationFailureHandler.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.server.authorization.web.authentication;
+
+import java.io.IOException;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.server.ServletServerHttpResponse;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.OAuth2Error;
+import org.springframework.security.web.authentication.AuthenticationFailureHandler;
+
+/**
+ * 	A default implementation of an {@link AuthenticationFailureHandler} used for handling an {@link OAuth2AuthenticationException}
+ * 	and returning the {@link OAuth2Error Error Response}.
+ *
+ * @author Dmitriy Dubson
+ * @since 1.2.0
+ * @see AuthenticationFailureHandler
+ */
+public class DefaultAuthenticationFailureHandler implements AuthenticationFailureHandler {
+	private final HttpMessageConverter<OAuth2Error> errorHttpResponseConverter;
+
+	public DefaultAuthenticationFailureHandler(HttpMessageConverter<OAuth2Error> errorHttpResponseConverter) {
```

Comment:
Let's remove the constructor in favour of a setter (`setErrorResponseConverter()`) and default the variable to `OAuth2ErrorHttpMessageConverter`